### ISSUE TITLE
Mathematical expressions marked incorrectly when uppercase variables are entered as part of an equation

### ISSUE THREAD

**Issue by georgestagg** [numbas/Numbas#739]
To reproduce:

* Add a mathematical expression type to a question
* Set the correct answer to "x=(y-B)/A"
* Run the exam, and try entering either "x=(y-B)/A" or "x=(y-b)/a"

In this case neither answer is accepted, the correct behaviour should probably be that either answer is accepted.

**Comment by tgreenhalgh** [comment]
I believe https://github.com/numbas/Numbas/pull/740 fixes this issue.
also, it's been a while, so not 100% sure I did the PR correctly, please let me know!

**Comment by christianp** [comment]
@tgreenhalgh Thanks for this, but it didn't really fix the issue. I found the point at which Numbas failed to apply case-insensitivity, and fixed that.

For reference, there are two ways Numbas can load a question: through XML, or JSON. XML is used for compiled exams, while JSON is used inside the editor. If we were starting from scratch we'd just use JSON, but the XML is what we started with and I don't want to break something that's currently working.
